example-frontend/pages/autoOffsetExactInToken.tsx

```diff
@@ -0,0 +1,85 @@
+import { useProvider, useSigner } from "wagmi";
+import { useState } from "react";
+import { FormatTypes, Interface, parseUnits } from "ethers/lib/utils";
+import { ethers } from "ethers";
+import OffsetHelper from "../../deployments/polygon/OffsetHelper.json";
+import ERC20 from "../abis/ERC20.json";
+
+export default function AutoOffsetExactInToken() {
+  // input data for the autoOffsetExactInToken function
+  const poolAddress = "0xD838290e877E0188a4A44700463419ED96c16107"; // Polygon - NCT
+  const depositedToken = "0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174"; // Polygon - USDC
+  const amount = parseUnits("0.0001", 6);
+  const [tx, setTx] = useState("");
+
+  // get signer & provider
+  const { data: signer } = useSigner();
+  const provider = useProvider();
+  // create contract for approve function of the ERC20 token
+  const iface = new Interface(ERC20.abi);
+  iface.format(FormatTypes.full);
+  const depositedTokenContract = new ethers.Contract(
+    depositedToken,
+    iface,
+    signer || provider
+  );
+
+  // create OffsetHelper contract
+  const offsetHelper = new ethers.Contract(
+    OffsetHelper.address,
+    OffsetHelper.abi,
+    signer || provider
+  );
+  const offset = async () => {
+    try {
+      // approve spending of deposited tokens
+      await (
+        await depositedTokenContract.approve(OffsetHelper.address, amount)
+      ).wait();
+
+      // retire carbon credits if the user already owns a Toucan pool token e.g., NCT,
+      const result = await offsetHelper.autoOffsetExactInToken(
+        depositedToken,
+        poolAddress,
+        amount
+      );
+      setTx(result.hash);
+    } catch (error) {
```

Comment:
@GigaHierz there is one more thing I would like to mention at this PR, can you make sure there is an alert or something when not logged in. As otherwise if not logged in, the respective functions are failing silently.

Reply:
can you please elaborate or provide an example? this message is not clear to me

Reply:
@GigaHierz sure, go to the homepage and click on "autoOffsetPoolToken" (redirects to the autoOffsetPoolToken button), try to click it without being logged with a wallet. Nothing happens unless you inspect the console.

Suggestion: Show an alert `alert("Please log in.")` as a simple measure.

Reply:
now I re-read the message and it makes sense.. I just didn't know what was not logged in.. like maybe the lucky numbers? and auction? the right answer? I just didn't know...

Reply:
Haha, communication is the hardest part for sure.